### PR TITLE
POM_RF | topsWomenPage.spec | TC 05.1.4_02 Women/Tops/Display mode

### PR DESCRIPTION
**Pull request by mlimau**
https://trello.com/c/gVbMRgx7/702-pomrf-topswomenpagespec-tc-051402-women-tops-display-modeverify-list-mode?filter=member:marinauge

### CODE REVIEW

File: tests/test_page_objects/topsWomenPage.spec.js [hunk 1]
```diff
@@ -73,4 +75,33 @@ test.describe("topWomenPage.spec", () => {
 
     expect(expectedNumberJacketItems).toEqual(actualNumberJacketItems);
   });
+
+  test('TC 05.1.4_01 Women/Tops/Display mode can be changed, visible', async ({
+    page
+  }) => {
+    const homePage = new HomePage(page)
+    const womenPage = new WomenPage(page)
```
Comment (verafes) [inline tests/test_page_objects/topsWomenPage.spec.js:83]: Could you review yesterday's Maria Lecture and update your code accordingly?
Our  approach is to instantiate **only one** object for the home page, with other pages accessed through chaining from homePage." like this: 
const homePage = new HomePage(page)
await homePage.open()
const womenPage =a wait homePage.hoverWomenLink()
const topsWomenPage = await womenPage.clickWomenTopsLink()

Comment (mlimau) [reply tests/test_page_objects/topsWomenPage.spec.js:83]: Do you mean to create a method 'hoverWomenLink' on womenPage POM file ?

Comment (BabintsevaS) [reply tests/test_page_objects/topsWomenPage.spec.js:83]: @mlimau I did refactoring for all "topsWomenPage.spec" file according to yesterday's Maria Lecture, update your brunch and you can see that.

Comment (verafes) [reply tests/test_page_objects/topsWomenPage.spec.js:83]: No, in your code of his test, 3 instances were created first. But we need to create only one HomePage object. 
just update 82-88 with the following:
const homePage = new HomePage(page)
await homePage.open()
const womenPage = await homePage.hoverWomenLink()
const topsWomenPage = await womenPage.clickWomenTopsLink()

Comment (verafes) [reply tests/test_page_objects/topsWomenPage.spec.js:83]: The refactoring of "topsWomenPage.spec is done perfectly. Thank you

**Review by teya-teya — DISMISSED** [review]

File: tests/test_page_objects/topsWomenPage.spec.js (hunk 1 above)
Comment (mlimau) [reply tests/test_page_objects/topsWomenPage.spec.js]: Updated

File: tests/test_page_objects/topsWomenPage.spec.js [hunk 2]
```diff
@@ -1,12 +1,28 @@
...
   MY_WISH_LIST_EMPTY_MESSAGE,
   TOPS_WOMEN_PAGE_END_POINT,
   JACKET_ITEMS,
 } from "../../helpers/testData"; 
+import { MODE_GRID_ACTIVE_ATTR_CLASS, MODE_LIST_ACTIVE_ATTR_CLASS } from '../../helpers/testWomenData'
+import MainMenuPage from "../../page_objects/mainMenu";
+
+const setup = async (page) => {
+  const homePage = new HomePage(page);
+  const womenPage = new WomenPage(page);
```
Comment (verafes) [inline tests/test_page_objects/topsWomenPage.spec.js:17]: We do not need 17, 18, 21, 22 lines in before method anymore.

**Review by IrenaKh95 — APPROVED** [review]

File: tests/test_page_objects/topsWomenPage.spec.js (hunk 2 above)
Comment (mlimau) [reply tests/test_page_objects/topsWomenPage.spec.js]: got it, but в моём тесте этого нет.,это уже нужно создавать карточку на рефакторинг и делать отделанный PR

**Review by verafes — COMMENTED** [review]
последний прогон CI показывает что как бы лишняя скобка на 68 строке в topsWomenPage. Нужно попробовать запустить тесты локально

File: tests/test_page_objects/topsWomenPage.spec.js (hunk 2 above)
Comment (verafes) [reply tests/test_page_objects/topsWomenPage.spec.js:17]: Ок , снимаю свое замечание, бывает что reformat code подсвечивает старые тесты как новые.

**Review by verafes — APPROVED** [review]

**Pull request closed without merging**